volumes/miovision/sql/function-aggregate-volumes_15min_tmc.sql

```diff
@@ -1,163 +1,71 @@
-			lead(min(datetime_bin)) OVER w AS next_start,
-			(EXTRACT(minutes FROM MAX(datetime_bin) - MIN(datetime_bin))::INT+1) AS span,
-			MIN(volume_uid) AS a_volume_uid,
-			CASE WHEN  COUNT(DISTINCT datetime_bin) < 15 --If there is at least 1 missing one-minute bin in the current 15-min bin
-				/*and at least one one-minute bin in the following 2 minute*/
-				AND lead(min(datetime_bin)) OVER w <= datetime_bin_15(datetime_bin) + interval '16 minutes'
-				/*and at least one one-minute bin in the preceding 2 minute*/
-				AND lag(max(datetime_bin)) OVER w  >= datetime_bin_15(datetime_bin) - interval '2 minutes' 
-				THEN FALSE 
-				ELSE NULL 
-				END AS interpolated
-		FROM miovision_api.volumes A
-		INNER JOIN miovision_api.intersection_movements	m  --Make sure movement is valid.
-		 											USING (intersection_uid, classification_uid,leg,movement_uid)
-		WHERE datetime_bin >= start_date - INTERVAL '2 hours' AND datetime_bin < end_date 
-		AND volume_15min_tmc_uid IS NULL
-		GROUP BY intersection_uid, datetime_bin_15
-		WINDOW w AS (PARTITION BY intersection_uid)
-	)
-
-	INSERT INTO 	bins
-	SELECT 		intersection_uid,
-			datetime_bin_15,
-			avail_minutes,
-			start_time,
-			end_time,
-			span,
-			interpolated,
-			a_volume_uid
-	FROM 		bin_grouping
-	WHERE avail_minutes>5;
-
-
-	UPDATE bins SET interpolated = FALSE WHERE (EXTRACT(minutes FROM end_time - start_time)+1) > avail_minutes AND avail_minutes < 15;
-	UPDATE bins SET interpolated = TRUE WHERE interpolated IS NULL AND avail_minutes < 15;
-
-	UPDATE bins
-	SET 	end_time = CASE
-		WHEN datetime_bin = start_time THEN end_time - INTERVAL '1 minute' ELSE end_time
-		END,
-		start_time = CASE
-		WHEN datetime_bin + INTERVAL '14 minutes' = end_time THEN start_time + INTERVAL '1 minute' ELSE start_time
-		END,
-		span =  CASE
-		WHEN datetime_bin = start_time THEN span - 1
-		WHEN datetime_bin + INTERVAL '14 minutes' = end_time THEN span -1
-		WHEN datetime_bin = start_time 
-			AND datetime_bin + INTERVAL '14 minutes' = end_time THEN span - 2 
- 		ELSE span END
-		WHERE interpolated = TRUE ;
-	RAISE NOTICE '% Interpolation finished', timeofday();
+INSERT INTO miovision_api.volumes_15min_tmc(intersection_uid, datetime_bin, classification_uid, leg, movement_uid, volume)
+
+SELECT 
+m.intersection_uid,
+dt.datetime_bin,
+m.classification_uid,
+m.leg,
+m.movement_uid,
+CASE WHEN un.accept = FALSE THEN NULL ELSE (COALESCE(SUM(A.volume), 0)) END AS volume
+FROM miovision_api.intersection_movements_new m
+--Cross product of legal movement for cars, bikes, and peds and the bins to aggregate
+CROSS JOIN generate_series(start_date - interval '1 hour', end_date - interval '1 hour 15 minutes', INTERVAL '15 minutes') AS dt(datetime_bin)
+--To get 1min bins
+LEFT JOIN miovision_api.volumes A
+	ON A.intersection_uid = m.intersection_uid AND A.classification_uid = m.classification_uid
+	AND A.leg = m.leg AND A.movement_uid = m.movement_uid
+	AND dt.datetime_bin = datetime_bin_15(A.datetime_bin)
```

Comment:
Join on a transformation of a timestamp is going to be slow do datetime_bin ≥ datetime_bin15 AND datetime_bin < datetime_bin15 + interval '15 minutes'